game/src/plugins/skills/fishing/src/fishing.plugin.kts

```diff
@@ -0,0 +1,133 @@
+import org.apollo.cache.def.ItemDefinition
+import org.apollo.game.action.DistancedAction
+import org.apollo.game.message.impl.NpcActionMessage
+import org.apollo.game.model.Position
+import org.apollo.game.model.entity.Player
+import org.apollo.game.model.entity.Skill
+import org.apollo.game.plugin.skills.fishing.*
+import kotlin.properties.Delegates
+import java.util.Random
+
+class FishingAction: DistancedAction<Player> {
+
+    var spot: FishingSpot by Delegates.notNull()
```

Comment:
`by Delegates.notNull()` is unnecessary.  Use `val` instead of `var` and assign once in the constructor (val = `immutable`, i.e. can't be changed.  var = `mutable`).